src/wallet/did_wallet/did_wallet.py

```diff
@@ -0,0 +1,760 @@
+            additions=spend_bundle.additions(),
+            removals=spend_bundle.removals(),
+            wallet_id=self.wallet_state_manager.main_wallet.wallet_info.id,
+            sent_to=[],
+            trade_id=None,
+        )
+        did_record = TransactionRecord(
+            confirmed_at_index=uint32(0),
+            created_at_time=uint64(int(time.time())),
+            to_puzzle_hash=did_coin.puzzle_hash,
+            amount=uint64(did_coin.amount),
+            fee_amount=uint64(0),
+            incoming=True,
+            confirmed=False,
+            sent=uint32(10),
+            spend_bundle=None,
+            additions=spend_bundle.additions(),
+            removals=spend_bundle.removals(),
+            wallet_id=self.wallet_info.id,
+            sent_to=[],
+            trade_id=None,
+        )
+        await self.standard_wallet.push_transaction(regular_record)
+        await self.standard_wallet.push_transaction(did_record)
+        return self
+
+    @staticmethod
+    async def create(
+        wallet_state_manager: Any,
+        wallet: Wallet,
+        wallet_info: WalletInfo,
+        name: str = None,
+    ):
+        self = DIDWallet()
+
+        if name:
+            self.log = logging.getLogger(name)
+        else:
+            self.log = logging.getLogger(__name__)
+
+        self.wallet_state_manager = wallet_state_manager
+        self.wallet_info = wallet_info
+        self.standard_wallet = wallet
+        self.did_info = DIDInfo.from_bytes(hexstr_to_bytes(self.wallet_info.data))
+        self.base_puzzle_program = None
+        self.base_inner_puzzle_hash = None
+        return self
+
+    async def get_confirmed_balance(self) -> uint64:
+        record_list: Set[
+            WalletCoinRecord
+        ] = await self.wallet_state_manager.wallet_store.get_unspent_coins_for_wallet(
+            self.wallet_info.id
+        )
+
+        amount: uint64 = uint64(0)
+        for record in record_list:
+            parent = await self.get_parent_for_coin(record.coin)
+            if parent is not None:
+                amount = uint64(amount + record.coin.amount)
+
+        self.log.info(f"Confirmed balance for did wallet is {amount}")
+        return uint64(amount)
+
+    async def get_unconfirmed_balance(self) -> uint64:
+        confirmed = await self.get_confirmed_balance()
+        unconfirmed_tx: List[
+            TransactionRecord
+        ] = await self.wallet_state_manager.tx_store.get_unconfirmed_for_wallet(
+            self.wallet_info.id
+        )
+        addition_amount = 0
+        removal_amount = 0
+
+        for record in unconfirmed_tx:
+            if record.incoming:
+                addition_amount += record.amount
+            else:
+                removal_amount += record.amount
+
+        result = confirmed - removal_amount + addition_amount
+
+        self.log.info(f"Unconfirmed balance for did wallet is {result}")
+        return uint64(result)
+
+    async def select_coins(
```

Comment:
Why does this class need a copy of this function? Is there something special about how it selects coins that's different from other wallets?